featuretools/utils/gen_utils.py

```diff
@@ -91,6 +91,43 @@ def import_or_raise(library, error_msg):
         raise ImportError(error_msg)
 
 
+def import_or_none(library):
+    '''
+    Attemps to import the requested library.
+
+    Args:
+        library (str): the name of the library
+    Returns: the library if it is installed, else None
+    '''
+    try:
+        return importlib.import_module(library)
+    except ImportError:
+        return None
+
+
 def camel_to_snake(s):
     s = re.sub('(.)([A-Z][a-z]+)', r'\1_\2', s)
     return re.sub('([a-z0-9])([A-Z])', r'\1_\2', s).lower()
+
+
+def is_instance(obj, modules, classnames):
```

Comment:
As an alternate approach here would it make more sense to structure this to accept a list of strings of the form "module.classname" directly, instead of having to parse/combine two separate lists for modules and classnames? That way the signature would look a little more like the built-in `isinstance` function 

You could then split the string that is passed and get the module with `globals().get(module_name)`.

So, call the function like this instead: `is_instance(df, ("ks.DataFrame", "dd.DataFrame"))`

Not sure if that is better or not, but just throwing it out for thought.

Reply:
I like that this approach mimics the built-in `isinstance`, but I think in order for it to work we'd need to import everything directly into the utils file (since `globals()` would return the function's global scope, not the callees'). To me, that's less than ideal since it removes some of the flexibility, and could possibly introduce namespace issues. We could get around this by adding the callees global dictionary as an argument to the function, so calling it might look something like this: `is_instance(df, ('ks.DataFrame', 'dd.DataFrame'), globals())`, but I'm not sure this is a great solution either.

Reply:
I wasn't thinking through the scoping issues enough when I made the suggestion. I agree that passing in the `globals()` from the calling scope isn't a great solution. So far, I haven't thought of a way to make my proposal work cleanly, so with that sticking with your original implementation might be the best approach here.

Reply:
what if we leveraged the fact that `ks` will be `None` if koalas is not installed and that python stops evaluating an "and" statement after the first false to make checks like this:

```python
if ks and isinstance(df, ks.DataFrame):
````
if `ks` is `None` the `isinstance` code won't be run, so we don't need to worry about it